lib/src/current_weather/data/repository/current_weather_repository.dart

```diff
@@ -20,11 +21,24 @@ class CurrentWeatherRepository implements CurrentWeatherRepositoryInterface {
     final connection = await connectivity.checkConnectivity();
     final hasInternet = connection != ConnectivityResult.none;
     if (!hasInternet) {
+      final currentWeather = getCurrentWeatherFromDb();
+      if (currentWeather != null) return currentWeather;
       throw const NoInternetAvailableException();
-    } //TODO implement Local;
+    }
 
     final result = await api.current(location.cityName);
     final model = CurrentWeatherDto.fromJson(result);
+    saveCurrentWeather(model.toEntity());
     return model.toEntity();
   }
+
+  @override
+  IsarCurrentWeatherDto saveCurrentWeather(CurrentWeather currentWeather) {
+    return local.saveCurrentWeather(currentWeather);
+  }
+
+  @override
+  CurrentWeather? getCurrentWeatherFromDb() {
+    return local.getCurrentWeatherFromDb();
+  }
```

Comment:
Great job! But these two methods - `saveCurrentWeather` and `getCurrentWeatherFromDb` should be private and shouldn't leak into the interface, either.

Also, why do we return a `CurrentWeather?`? Why is `null` an option? If anything goes wrong, I'd rather `throw` an exception.